From d8b300ebd8e049e0bf1b87dab9091e047cc4b3bc Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 19 Oct 2023 15:36:01 +0200
Subject: [PATCH] Update PythonBinding.h

---
 src/python/PythonBinding.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/python/PythonBinding.h b/src/python/PythonBinding.h
index ed273a1..8812e72 100644
--- a/src/python/PythonBinding.h
+++ b/src/python/PythonBinding.h
@@ -64,7 +64,7 @@ NAMESPACE_BEGIN(detail)
 			return pybind11::make_tuple(src[0], src[1], src[2]).release();
         }
 
-		PYBIND11_TYPE_CASTER(atomicrex::Vector_3<T>, _("Vector3<") + make_caster<T>::name() + _(">"));
+		PYBIND11_TYPE_CASTER(atomicrex::Vector_3<T>, _("Vector3<") + make_caster<T>::name + _(">"));
     };	
 
 	/// Automatic Python <--> Point3 conversion
@@ -84,7 +84,7 @@ NAMESPACE_BEGIN(detail)
 			return pybind11::make_tuple(src[0], src[1], src[2]).release();
         }
 
-		PYBIND11_TYPE_CASTER(atomicrex::Point_3<T>, _("Point3<") + make_caster<T>::name() + _(">"));
+		PYBIND11_TYPE_CASTER(atomicrex::Point_3<T>, _("Point3<") + make_caster<T>::name + _(">"));
     };		
 
 	/// Automatic Python <--> Matrix3 conversion
@@ -114,7 +114,7 @@ NAMESPACE_BEGIN(detail)
 				src.elements()).release();
         }
 
-		PYBIND11_TYPE_CASTER(atomicrex::Matrix_3<T>, _("Matrix3<") + make_caster<T>::name() + _(">"));
+		PYBIND11_TYPE_CASTER(atomicrex::Matrix_3<T>, _("Matrix3<") + make_caster<T>::name + _(">"));
     };
 
 NAMESPACE_END(detail)
